src/components/__tests__/rating.js

```diff
@@ -0,0 +1,31 @@
+import React from 'react'
+import {render, screen, loginAsUser, waitFor} from 'test/app-test-utils'
+import userEvent from '@testing-library/user-event'
+import {buildBook, buildListItem} from 'test/generate'
+import * as booksDB from 'test/data/books'
+import * as listItemsDB from 'test/data/list-items'
+import {Rating} from '../rating'
+
+async function renderRating({rating = 0} = {}) {
+  const book = await booksDB.create(buildBook())
+  const user = await loginAsUser()
+  const listItem = await listItemsDB.create({
+    ...buildListItem({owner: user, book}),
+    rating,
+  })
+  const utils = await render(<Rating listItem={listItem} />, {user})
+  return {...utils, book, user, listItem}
+}
+
+test('it updates the rating', async () => {
+  const {listItem, rerender} = await renderRating()
+  const firstStar = screen.getByLabelText('1 star')
+
+  userEvent.click(firstStar)
+
+  // needed because of react-query
+  await waitFor(() => {})
+  const updatedListItem = await listItemsDB.read(listItem.id)
+  rerender(<Rating listItem={updatedListItem} />)
+  expect(screen.getByLabelText('1 star')).toBeChecked()
```

Comment:
This one's actually a bit tricky. What we really care about is that the right radio gets selected and the database is updated (eventually).

Unfortunately, `waitFor` doesn't handle async stuff within the callback function (this may be a useful feature to build in though eventually).

So here's what I came up with:

```suggestion
  expect(screen.getByLabelText('1 star')).toBeChecked()
  let updatedListItem
  do {
    updatedListItem = await listItemsDB.read(listItem.id)
  } while (updateListItem.rating !== 1)
```

I don't love it, but I think it's the best way to get the confidence in the things we want.